projects/utils-maps/src/lib/ol/utils-ol-mapstate.ts

```diff
@@ -0,0 +1,204 @@
+  const transformedCenter = transform(map.getView().getCenter(), srcProjection, dstProjection);
+  return transformedCenter;
+}
+
+
+/**
+  *
+  * @param features: olFeature[]
+  * @param geographic: boolean
+  * @returns olExtend: [minX, minY, maxX, maxY]
+  */
+export function getFeaturesExtent(map: Map, features: Feature<any>[], geographic?: boolean) {
+  const extent: any = features[0].getGeometry().getExtent().slice(0);
+  features.forEach((feature) => {
+    olExtend(extent, feature.getGeometry().getExtent());
+  });
+  if (geographic) {
+    const projection = getProjection('EPSG:4326');
+    const transformedExtent = transformExtent(extent, map.getView().getProjection().getCode(), projection);
+    return (transformedExtent as Extent);
+  } else {
+    return extent;
+  }
+}
+
+/**
+ * @param geographic: boolean
+ * @returns olExtend: [minX, minY, maxX, maxY]
+ */
+export function getCurrentExtent(map: Map, geographic?: boolean) {
+  const projection = (geographic) ? getProjection('EPSG:4326') : getProjection(DEFAULT_PROJECTION);
+  const extent = map.getView().calculateExtent();
+  const transformedExtent = transformExtent(extent, map.getView().getProjection().getCode(), projection);
+  return (transformedExtent as Extent);
+}
+
+
+export function setZoom(map: Map, zoom: number, notifier?: 'map' | 'user') {
+  const view = map.getView();
+  view.setZoom(zoom);
+}
+
+/** USED in map-ol.component */
+export function getZoom(map: Map): number {
+  return map.getView().getZoom();
+}
+
+export function zoomInOut(map: Map, value: '-' | '+') {
+  const view = map.getView();
+  if (!view) {
+    // the map does not have a view, so we can't act
+    // upon it
+    return;
+  }
+  const delta = 1, duration = 250;
+  const currentZoom = view.getZoom();
+  if (currentZoom !== undefined) {
+    const newZoom = view.getConstrainedZoom(currentZoom + delta);
+    if (duration > 0) {
+      if (view.getAnimating()) {
+        view.cancelAnimations();
+      }
+      view.animate({
+        zoom: newZoom,
+        duration,
+        easing: easeOut
+      });
+    } else {
+      view.setZoom(newZoom);
+    }
+  }
+}
+
+
+
+export function getMapProjection(map: Map) {
+  return map.getView().getProjection();
+}
+
+
+/**
+ * vector layers will be reprojected automatically
+ * wms layers will be updated with corresponding proj def in the requests.
+ * for other raster layers and for those wms layers whose backend does not support target projection, please
+ * define initial(default) layer projection, so openlayers will reproject on the client side
+ * projection is proj~ProjectionLike
+ */
+export function setMapProjection(map: Map, projection: Projection | string, viewOptions?: ViewOptions, cb?: (projection: Projection) => void) {
+  let epsg = DEFAULT_PROJECTION;
+  let view = null;
+  if (projection) {
+    let newViewOptions: ViewOptions = {};
+    if (viewOptions) {
+      newViewOptions = viewOptions;
+      newViewOptions.minResolution = undefined;
```

Comment:
> `newViewOptions.minResolution = undefined;`
Here we overwrite the user-provided `viewOptions` with undefined. I'm honestly not certain why we need `minResolution` et. al. to be undefined - do you know, @boeckMt ? 
I'll add a comment explaining that reason then :)

Reply:
When we copy the old view settings, the resolutions are not valid for the new projections, especially if the new and old projections have different unit systems, so they must be undefined, otherwise you can't initialize the new view... please correct it if my logic was wrong

Reply:
Aaaahhh!
That explains a lot. 

That line of thought makes perfect sense inside `MapOlService.setProjection`, because there we unset the `minResolution` of the *current* viewOptions `this.viewOptions`.

Here, however, we created a function `utils-ol-mapstate.ts/setMapProjection` that is *independent* of the current viewOptions.  Instead, the user is asked to provide viewOptions himself - so overwriting the users instructions here might be confusing. 

We overwrite quite a few of the viewOptions-properties here: 
 - minResolution
 - maxResolution
 - resolution
 - resolutions
 - center
 - zoom 

I think it would be better to not let the user provide a viewOptions-object at all. I think the actual intention here is to use the *old* viewOptions as a base for constructing the new ones, analogous to what we do in `MapOlService`. In that case, it would make sense to remove `viewOptions` as an argument from this function and instead obtain the `oldViewOptions`. Openlayers doesn't provide a public method to retrieve those options, I believe, but 
```const oldViewOptions = map.getView().getUpdatedOptions_()```
seems to do the job.

Reply:
I have to say that most of the functions which are not in `utils-ol-layers.ts` are only copyed from `MapOlService` so the need more adjustments!!!

My Idea was to outsource as much as possible from `MapOlService` to the 'Utils' but to reuse it we have to create the function in a way that we later can implement all the Stuff in `MapOlService` again without duplicating to much code.